qt-models/maplocationmodel.cpp

```diff
@@ -8,6 +8,9 @@ const char *MapLocation::PROPERTY_NAME_NAME       = "name";
 
 MapLocation::MapLocation()
 {
+	m_uuid = 0;
+	m_name = "";
+	m_coordinate = QGeoCoordinate();
```

Comment:
This one is even worse. The old code default-constructed m_coordinate. The new code:
1) Default constructs m_coordinate
2) Default constructs a temporary QGeoCoordinate
3) Assigns the latter to the former
Unless the default-constructor of QGeoCoorindate is inlined, this kind of things can give pretty terrible code.

That's why generally for complex types (notably those with constructors) initialization lists are preferred: They will call exactly one constructor. Assignment in the body will construct two objects and assign (unless the compiler is smart enough to do copy elision, etc).